src/diff/children.js

```diff
@@ -329,25 +650,45 @@ function findMatchingIndex(
 	let x = skewedIndex - 1;
 	let y = skewedIndex + 1;
 	let oldVNode = oldChildren[skewedIndex];
+	// We only need to perform a search if there are more children
+	// (remainingOldChildren) to search. However, if the oldVNode we just looked
+	// at skewedIndex was not already used in this diff, then there must be at
+	// least 1 other (so greater than 1) remainingOldChildren to attempt to match
+	// against. So the following condition checks that ensuring
+	// remainingOldChildren > 1 if the oldVNode is not already used/matched. Else
+	// if the oldVNode was null or matched, then there could needs to be at least
+	// 1 (aka `remainingOldChildren > 0`) children to find and compare against.
+	let shouldSearch =
+		remainingOldChildren > (oldVNode != null && !oldVNode._matched ? 1 : 0);
```

Comment:
TODO: verify terser inlines this nicely and removes the unnecessary variable for size purposes.